modules/line_descriptor/src/LSDDetector.cpp

```diff
@@ -148,48 +148,49 @@ void LSDDetector::detectImpl( const Mat& imageSrc, std::vector<KeyLine>& keyline
   cv::Ptr<cv::LineSegmentDetector> ls = cv::createLineSegmentDetector( cv::LSD_REFINE_ADV );
 
   /* prepare a vector to host extracted segments */
-  std::vector<std::vector<cv::Vec4i> > lines_lsd;
+  std::vector<std::vector<cv::Vec4f> > lines_lsd;
 
   /* extract lines */
   for ( int i = 0; i < numOctaves; i++ )
   {
-    std::vector<Vec4i> octave_lines;
+	 std::vector<Vec4f> octave_lines;
     ls->detect( gaussianPyrs[i], octave_lines );
     lines_lsd.push_back( octave_lines );
   }
 
   /* create keylines */
   int class_counter = -1;
-  for ( int j = 0; j < (int) lines_lsd.size(); j++ )
+  for ( int octaveIdx = 0; octaveIdx < (int) lines_lsd.size(); octaveIdx++ )
   {
-    for ( int k = 0; k < (int) lines_lsd[j].size(); k++ )
+	 float octaveScale = pow( (float)scale, octaveIdx );
+	 for ( int k = 0; k < (int) lines_lsd[octaveIdx].size(); k++ )
     {
       KeyLine kl;
-      cv::Vec4i extremes = lines_lsd[j][k];
+		cv::Vec4f extremes = lines_lsd[octaveIdx][k];
 
       /* check data validity */
-      checkLineExtremes( extremes, gaussianPyrs[j].size() );
+		checkLineExtremes( extremes, gaussianPyrs[octaveIdx].size() );
 
       /* fill KeyLine's fields */
-      kl.startPointX = (float) extremes[0];
-      kl.startPointY = (float) extremes[1];
-      kl.endPointX = (float) extremes[2];
-      kl.endPointY = (float) extremes[3];
-      kl.sPointInOctaveX = (float) extremes[0];
-      kl.sPointInOctaveY = (float) extremes[1];
-      kl.ePointInOctaveX = (float) extremes[2];
-      kl.ePointInOctaveY = (float) extremes[3];
-      kl.lineLength = (float) sqrt( pow( extremes[0] - extremes[2], 2 ) + pow( extremes[1] - extremes[3], 2 ) );
+		kl.startPointX = extremes[0] * octaveScale;
```

Comment:
Just a minor style remark: Please make sure that your intendation is set to spaces, not to tabs.